src/test/java/bio/terra/common/iam/SamUserFactoryTest.java

```diff
@@ -0,0 +1,64 @@
+package bio.terra.common.iam;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import bio.terra.common.exception.UnauthorizedException;
+import org.broadinstitute.dsde.workbench.client.sam.ApiException;
+import org.broadinstitute.dsde.workbench.client.sam.api.UsersApi;
+import org.broadinstitute.dsde.workbench.client.sam.model.UserStatusInfo;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpStatus;
+
+@Tag("unit")
+public class SamUserFactoryTest {
+
+  private static final SamUser SAM_USER =
+      new SamUser("test@example.com", "Subject", new BearerToken("0123.456-789AbCd"));
+  private static final String SAM_BASE_PATH = "not_real";
+
+  @Test
+  public void enabledUser() throws ApiException {
+    SamUserFactory factory = spy(new SamUserFactory(new BearerTokenFactory()));
+    UsersApi usersApi = mock(UsersApi.class);
+    when(factory.createUsersApi(SAM_USER.getBearerToken(), SAM_BASE_PATH)).thenReturn(usersApi);
```

Comment:
The main downside to `spy()` is that mocking a spied object method is different a mocked object method. Writing the mock this way will result in a call to to the real object's `factory.createUsersApi()` method. Instead you need to mock using this form:
```
    doReturn(usersApi).when(factory).createUsersApi(SAM_USER.getBearerToken(), SAM_BASE_PATH);
```